src/Umbraco.Core/Scoping/ScopeProvider.cs

```diff
@@ -181,51 +182,31 @@ private static void SetCallContextObject(string key, IInstanceIdentifiable value
             }
         }
 
-        // this is for tests exclusively until we have a proper accessor in v8
-        internal static Func<IDictionary> HttpContextItemsGetter { get; set; }
 
-        private static IDictionary HttpContextItems => HttpContextItemsGetter == null
-            ? HttpContext.Current?.Items
-            : HttpContextItemsGetter();
-
-        public static T GetHttpContextObject<T>(string key, bool required = true)
+        private T GetHttpContextObject<T>(string key, bool required = true)
             where T : class
         {
-            var httpContextItems = HttpContextItems;
-            if (httpContextItems != null)
-                return (T)httpContextItems[key];
-            if (required)
+
+            var result = (T) _requestCache.Get(key);
+
+            if (result is null && required)
+            {
                 throw new Exception("HttpContext.Current is null.");
-            return null;
+            }
+
+            return result;
+
         }
 
-        private static bool SetHttpContextObject(string key, object value, bool required = true)
+        private bool SetHttpContextObject(string key, object value, bool required = true)
         {
-            var httpContextItems = HttpContextItems;
-            if (httpContextItems == null)
-            {
-                if (required)
-                    throw new Exception("HttpContext.Current is null.");
-                return false;
-            }
-#if DEBUG_SCOPES
```

Comment:
did you want to remove this directive? I've never used it but never had to debug scopes before

Reply:
I'm a bit unsure how to port it.. Do you have any ideas?

Reply:
@bergmania I've pushed an update to this branch, see rev 898167f8c60f7ac093d4fe12205b5adf9506619f

Let me know what you think.